virttest/qemu_vm.py

```diff
@@ -3912,6 +3914,18 @@ def migrate(self, timeout=virt_vm.BaseVM.MIGRATE_TIMEOUT, protocol="tcp",
 
             logging.info("Migrating to %s", uri)
             self.monitor.migrate(uri)
+
+            if mig_inner_funcs:
+                for (func, param) in mig_inner_funcs:
+                    if func == "postcopy":
```

Comment:
Yep, this should allow to share the same types of functions. I'm still thinking on how to improve that to allow direct "vm.migrate(..., mig_inner_funcs=params.get("mig_inner_funcs"))" calls that would allow all tests to be always prepared to execute mig_inner_funcs without further changes. But that would require a text-based input or `ast.literal_eval`.

Actually if we decide to go with `ast.literal_eval` we can add the support even afterwards. So let's start with this and eventually we can improve later...

Reply:
Well actually you already do that in the migrate test:

https://github.com/dagrh/tp-qemu/commit/c67980ceb735f18cbbf90074aedc87e41c10dafc#diff-44f30088468ef7c6d55a3a1d9e942559R210

but IMO it'd be better to do the `ast.literal_eval` here in `qemu_vm` as that way the only change to all migrates jobs would be to add `params.get(...)` calls and that would be it.

Reply:
Still you could do that conditionally using `if not isinstance(mig_inner_funcs, (list, tuple)); mig_inner_funcs = ast.literal_eval(mig_inner_funcs)`. What do you think?

Reply:
The capabilities parameter that's already there doesn't have the literal_eval, so having them this way is at least consistent in adding it on the outside.